File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/property/detached-structure.ts
import { InspectionOrderProperty } from "../property";
import { InspectionOrderPropertyDetachedStructureOtherDetachedStructuresDetails } from "./detached-structure/InspectionOrderPropertyDetachedStructureOtherDetachedStructuresDetails";
import { InspectionOrderPropertyDetachedStructureOutbuildingDetails } from "./detached-structure/InspectionOrderPropertyDetachedStructureOutbuildingDetails";

export class InspectionOrderPropertyDetachedStructure {
    /*[Parent]*/
    inspectionOrderProperty?: InspectionOrderProperty;
    /*[Key]*/
    id?: string;
    outbuilding?: boolean;
    outbuildingDetails?: InspectionOrderPropertyDetachedStructureOutbuildingDetails[];
    outbuildingTypeOrConstruction?: string;
    outbuildingConcernOrComment?: string;
    otherDetachedStructure?: boolean;
    otherDetachedStructuresDetails?: InspectionOrderPropertyDetachedStructureOtherDetachedStructuresDetails[];
    public constructor(init?:Partial<InspectionOrderPropertyDetachedStructure>) {
        Object.assign(this, init);
    }
  }

File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderProperty/DetachedStructure.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderPropertyDetachedStructure
    {
        [Parent]
        public InspectionOrderProperty InspectionOrderProperty { get; set; }
        [Key]
        public Guid Id { get; set; }

        public bool Outbuilding { get; set; }
        public ICollection<InspectionOrderPropertyDetachedStructureOutbuildingDetails> OutbuildingDetails { get; set; }
        public string OutbuildingTypeOrConstruction { get; set; }
        public string OutbuildingConcernOrComment { get; set; }
        public bool OtherDetachedStructure { get; set; }
        public ICollection<InspectionOrderPropertyDetachedStructureOtherDetachedStructuresDetails> OtherDetachedStructuresDetails { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<InspectionOrderProperty>()
                .HasOne(a => a.DetachedStructure).WithOne(b => b.InspectionOrderProperty)
                .HasForeignKey<InspectionOrderPropertyDetachedStructure>()
                .HasConstraintName($"FK_{nameof(InspectionOrderProperty)}_{nameof(InspectionOrderPropertyDetachedStructure)}")
                .OnDelete(DeleteBehavior.Cascade);

            InspectionOrderPropertyDetachedStructureOutbuildingDetails.BuildModel(modelBuilder);

            InspectionOrderPropertyDetachedStructureOtherDetachedStructuresDetails.BuildModel(modelBuilder);
        }
    }
}